ps.go

```diff
@@ -84,3 +96,68 @@ func printContainers(ctx context.Context, clicontext *cli.Context, containers []
 	}
 	return w.Flush()
 }
+
+func containerStatus(ctx context.Context, c containerd.Container) string {
+	// Just in case, there is something wrong in server.
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
```

Comment:
Should this be in newClient() in client.go?

Reply:
If the one shim has dead lock issue, the task.Status will get stuck without timeout and `ps` also get stuck. So I add the magic 5 second here to prevent `ps action` getting stuck from one dead-lock-shim. WDYT?